File: xiang.li/53.kt
import kotlin.math.max
class Solution {
    fun maxSubArray(nums: IntArray): Int {
        for(i in 1 until nums.size) {
            nums[i] = max(nums[i], nums[i] + nums[i - 1])
            if(nums[i] > nums[0]) {
                nums[0] = nums[i]
            }
        }
        return nums[0]
    }
}

File: hang.liu/53.js
var maxSubArray = function(nums) {
  var result = nums[0];
  var sums = nums.slice();

  var length = sums.length;

  for (var i = 1; i < length; i++) {
    sums[i] = Math.max(sums[i], sums[i] + sums[i - 1]);
    result = Math.max(result, sums[i]);
  }

  return result;
};
